### ISSUE TITLE
To what degree can a Python function return type hint be customised?

### ISSUE THREAD

**Issue by MarkShackman** [doxygen/doxygen#10153]
(doxygen 1.9.7)

I'd like to be able to prevent a Python return type, generated from a type hint, from being linked*. 

For example, if I had:
`    def myMethod() -> myClass:`
inserting a % in front of the `myClass` still hyperlinks the returned class name. 

Is there any form of doxygen link processing available for Python type hints?

-------------------------------------
\* Not really true - I'd like to customise the link using a @ref tag, but that doesn't work either, hence am trying to prevent all linking of the returned class name.

**Comment by albert-github** [comment]
To see what you exactly want to do:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by MarkShackman** [comment]
[PythonTypeHintsIssue.zip](https://github.com/doxygen/doxygen/files/11893800/PythonTypeHintsIssue.zip)

I was hoping the % would be used to prevent the class name from being linked.

![image](https://github.com/doxygen/doxygen/assets/23169476/4158ada1-0118-47e8-b274-8411fd7dd0b9)

**Comment by albert-github** [comment]
Very nice small example and the image exactly shows what you intend 👍 

The problem here is that, to the best of my knowledge, now the statement:
```
def myMethod() -> %myClass:
```
is not valid python code anymore and otherwise the `%` has, probably, a special meaning in python.
The `%` in doxygen is indeed for disabling the linking in the comment. 

In python there are also normal usages of the `%` sign, like in formats etc.
In other languages there are also possibilities of usage the `%` e.g modulo in C / C++, reference to a structure element in Fortran.

I don't think it will be possible to prevent the link by means of `%` in this way

**Comment by MarkShackman** [comment]
I'm actually adding the % in a filter, rather than putting it in the Python code itself, so am not invalidating the language. I was hoping that all links are processed in doxygen similarly (respecting the %) even if it doesn't conform to the language.
------------------
The original problem was the duplicated class name shown in the screenshot. If another class's method returns a `myClass`, I'd have to specify a type hint of
`anotherClass() -> myClass.myClass` for doxygen to link it correctly.
Is there a way to remove the duplication?

**Comment by MarkShackman** [comment]
No, `myClass.myClass` isn't valid Python - I have to specify `\ref myClass.myClass "myClass"` to get the link in doxygen output.  Will look into using `cls.myClass` or `self.myClass`.

(Spend a fair few hours trying to resolve this, but haven't found a solution yet, so having to adapt the documentation to cope with it.)
